pkg/nmstate/nmstateconfig.go

```diff
@@ -0,0 +1,162 @@
+package nmstate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	assistedv1beta1 "github.com/openshift/assisted-service/api/v1beta1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Builder provides struct for the NMStateConfig object containing connection to
+// the cluster and the NMStateConfig definitions.
+type NmStateConfigBuilder struct {
+	// NMStateConfig definition. Used to create NMStateConfig object with minimum set of required elements.
+	Definition *assistedv1beta1.NMStateConfig
+	// Created NMStateConfig object on the cluster.
+	Object *assistedv1beta1.NMStateConfig
+	// API client to interact with the cluster.
+	apiClient *clients.Settings
+	// errorMsg is processed before NMStateConfig object is created.
+	errorMsg string
+}
+
+// NewBuilder creates a new instance of NMStateConfig Builder.
+func NewNmStateConfigBuilder(apiClient *clients.Settings, name, namespace string) *NmStateConfigBuilder {
```

Comment:
Please re-name to  NewConfigBuilder. You have nmstate in the name of the package. 
So if you need to trigger builder you'll use following call:
nmstate.NewConfigBuilder.